tests/runtime/in_podman_metrics.c

```diff
@@ -28,11 +28,11 @@
 #define DPATH_PODMAN_GARBAGE        FLB_TESTS_DATA_PATH "/data/podman/garbage"
 
 
-int check_metric(flb_ctx_t *ctx, flb_sds_t *name) {
+int check_metric(flb_ctx_t *ctx, char *name) {
```

Comment:
This is to fix following warning.

```
/home/taka/git/fluent-bit/tests/runtime/in_podman_metrics.c:47:24: warning: passing argument 1 of ‘strlen’ from incompatible pointer type [-Wincompatible-pointer-types]
   47 |             if (strlen(name) != 0 && strcmp(name, counter->opts.name) == 0)
      |                        ^~~~
      |                        |
      |                        char **
In file included from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack/pack_define.h:15,
                 from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack/pack.h:13,
                 from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack.h:19,
                 from /home/taka/git/fluent-bit/include/fluent-bit.h:23,
                 from /home/taka/git/fluent-bit/tests/runtime/in_podman_metrics.c:20:
/usr/include/string.h:407:35: note: expected ‘const char *’ but argument is of type ‘char **’
  407 | extern size_t strlen (const char *__s)
      |                       ~~~~~~~~~~~~^~~
/home/taka/git/fluent-bit/tests/runtime/in_podman_metrics.c:47:45: warning: passing argument 1 of ‘strcmp’ from incompatible pointer type [-Wincompatible-pointer-types]
   47 |             if (strlen(name) != 0 && strcmp(name, counter->opts.name) == 0)
      |                                             ^~~~
      |                                             |
      |                                             char **
In file included from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack/pack_define.h:15,
                 from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack/pack.h:13,
                 from /home/taka/git/fluent-bit/lib/msgpack-c/include/msgpack.h:19,
                 from /home/taka/git/fluent-bit/include/fluent-bit.h:23,
                 from /home/taka/git/fluent-bit/tests/runtime/in_podman_metrics.c:20:
/usr/include/string.h:156:32: note: expected ‘const char *’ but argument is of type ‘char **’
  156 | extern int strcmp (const char *__s1, const char *__s2)
      |                    ~~~~~~~~~~~~^~~~
```

Reply:
BTW, it would be disappeared the warning with using `flb_sds_t name` instead of `char *name` in this argument.
We might have to use `flb_sds_t` instead of raw type of `char *`. What do you think?

Reply:
In theory this could be a typo since `flb_sds_t` is a typedef to a `char *` but in  practice all the calls to this function are passing a `const char *` and the function doesn't treat the argument as a `flb_sds_t` (ie. `flb_sds_len` is not used) so the argument type should be `const char *` , otherwise it could cause issues when someone tries to use the `flb_sds_` functions (because this is a raw buffer with no sds header).

Reply:
OK. We can leave it as-is. I rechecked the passing pointers and they are all `const char*` types. So, leaving as-is is reasonable.